src/stcal/alignment/resample_utils.py

```diff
@@ -39,3 +41,211 @@ def calc_pixmap(in_wcs, out_wcs, shape=None):
     grid = grid_from_bounding_box(bb)
     transform_function = util.reproject(in_wcs, out_wcs)
     return np.dstack(transform_function(grid[0], grid[1]))
+
+
+def combine_sregions(sregion_list, det2world, intersect_footprint=None):
+    """
+    Combine s_regions from input models to compute the s_region for the resampled data.
+
+    Parameters
+    ----------
+    sregion_list : list of str
+        List of s_regions from input models.
+    det2world : `~astropy.modeling.Model`
+        WCS detector-to-world transform for the resampled data.
+        Must take in exactly two inputs (x, y) and return exactly two outputs (RA, Dec).
+        Must have a valid inverse transform.
+    intersect_footprint : np.ndarray, optional
+        Footprint of the output WCS in world coordinates, shape (N, 2).
+        If provided, the combined footprint from the input s_region list
+        will be intersected with this footprint.
+
+    Returns
+    -------
+    str
+        The combined s_region.
+    
+    Raises
+    ------
+    ValueError
+        If there is no overlap between the input s_regions and the intersection footprint.
+    """
+    footprints = np.array([util.sregion_to_footprint(sregion) for sregion in sregion_list])
```

Comment:
```suggestion
    footprints = np.array([util.sregion_to_footprint(sregion) for sregion in sregion_list])
```
Would you pull this conversion from strings to footprints outside of this function or make it conditional? That way if romancal wants to use this it can provide the the [footprints computed from L2 files](https://github.com/spacetelescope/romancal/blob/450348798730a7e11a88cb7383dd512ecdc0ccdf/romancal/resample/resample_utils.py#L77) without needing to first convert them to S_REGION strings (with a loss of precision).

Reply:
Yeah I can definitely do that.  What would be the most useful?

Would it make sense to have a method `combine_footprints_world(footprints, det2world)` that takes in a list of footprints and outputs a list of combined footprints?  And then make `_polygons_to_sregion` public, so if you did want to go all the way to an output s_region you could call that one?

Would it make sense to make the current `combine_footprints()` private, since it operates only in pixel space?

I think the easiest thing would be to make it a conditional but that solution seems clunky to me.

Reply:
ok to keep it consistent with `wcs_from_sregions` I just made it conditional.  How does https://github.com/spacetelescope/stcal/pull/407/commits/7f5ede5bd28c8b7934c2b90a3b7ce4fa4e81f236 look?  Unit test for it is here https://github.com/spacetelescope/stcal/pull/407/commits/a998995f19f9e3b1d615d848eadf7f3fa6512e13

Reply:
Looks good to me. Thanks!